caikit/runtime/service_generation/protoable.py

```diff
@@ -66,6 +77,11 @@ def handle_protoables_in_union(arg_type: Type) -> Type:
                 for union_val in typing.get_args(arg_type)
                 if is_protoable_type(union_val)
             ]
+            # handle union of lists in a separate way
+            if len(union_protoables) > 1 and all(
```

Comment:
Why is the logic here using `all`? What about `Union[List[int], int]`? I would think we'd want to support this too

Reply:
Added a test case to check for that condition as well (`Union[List], int`)